prov/tcp/src/xnet_progress.c

```diff
@@ -183,6 +179,69 @@ static int xnet_handle_truncate(struct xnet_ep *ep)
 	return 0;
 }
 
+static int xnet_queue_ack(struct xnet_ep *ep, uint8_t op, uint8_t op_data)
+{
+	struct xnet_xfer_entry *resp;
+
+	assert(xnet_progress_locked(xnet_ep2_progress(ep)));
+	assert(op == xnet_op_msg || op == xnet_op_cts);
+	resp = xnet_alloc_xfer(xnet_ep2_progress(ep));
+	if (!resp)
+		return -FI_ENOMEM;
+
+	resp->iov[0].iov_base = (void *) &resp->hdr;
+	resp->iov[0].iov_len = sizeof(resp->hdr.base_hdr);
+	resp->iov_cnt = 1;
+
+	resp->hdr.base_hdr.version = XNET_HDR_VERSION;
+	resp->hdr.base_hdr.op_data = op_data;
+	resp->hdr.base_hdr.op = op;
+	resp->hdr.base_hdr.size = sizeof(resp->hdr.base_hdr);
+	resp->hdr.base_hdr.hdr_size = (uint8_t) sizeof(resp->hdr.base_hdr);
+
+	resp->ctrl_flags = XNET_INTERNAL_XFER;
+	resp->context = NULL;
+
+	xnet_tx_queue_insert(ep, resp);
+	return FI_SUCCESS;
+}
+
+static int
+xnet_rts_matched(struct xnet_rdm *rdm, struct xnet_ep *ep,
+		 struct xnet_xfer_entry *rx_entry)
+{
+	struct xnet_conn *conn;
+	uint8_t cts_ctx;
+	int ret;
+
+	assert(xnet_progress_locked(xnet_rdm2_progress(rdm)));
+	if (!ep) {
+		if (xnet_get_conn(rdm, rx_entry->src_addr, &conn)) {
+			FI_WARN(&xnet_prov, FI_LOG_EP_DATA,
```

Comment:
compiler complained ret may be uninitialized here

Reply:
uhm, yeah, that's true.  I'll update.  I don't expect to ever go through this path, but it's theoretically possible